src/System.Memory/tests/ReadOnlyMemory/Equality.cs

```diff
@@ -84,6 +84,58 @@ public static void EmptyMemoryNotUnified()
             Assert.True(right.Equals(left));
         }
 
+        [Fact]
+        public static void EqualityThroughInterface_True()
+        {
+            int[] array = { 42, 43, 44, 45, 46 };
+            IEquatable<ReadOnlyMemory<int>> left = new ReadOnlyMemory<int>(array, 2, 3);
+            IEquatable<ReadOnlyMemory<int>> right = new ReadOnlyMemory<int>(array, 2, 3);
+
+            Assert.True(left.Equals(right));
```

Comment:
resolves to object.Equals(object) call. Again, I think the test for type mismatch should go here.